plonky2/src/gates/mod.rs

```diff
@@ -24,4 +24,5 @@ pub mod util;
 
 // Can't use #[cfg(test)] here because it needs to be visible to other crates.
 // See https://github.com/rust-lang/cargo/issues/8379
+#[cfg(any(feature = "rand", feature = "gate_testing", test))]
```

Comment:
Hm this doesn't seem quite right to me, like if rand is true but the others are false, then we don't need this right?

I think we can do `any(feature = "gate_testing", test)`, and then have `gate_testing` require `rand` in Cargo.toml.